Ginger/GingerCoreNET/Application Models/Delta/APIDelta/DeltaAPIModel.cs

```diff
@@ -0,0 +1,309 @@
+﻿using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Repository;
+using GingerCore.GeneralLib;
+using GingerCoreNET.Application_Models;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GingerCoreNET.Application_Models
+{
+    public class DeltaAPIModel
+    {
+        //private bool mNewSelected = false;
+        //public bool NewSelected
+        //{
+        //    get
+        //    {
+        //        return mNewSelected;
+        //    }
+        //    set
+        //    {
+        //        mNewSelected = value;
+        //        mExistingSelected = !value;
+        //    }
+        //}
+
+        //private bool mExistingSelected = false;
+        //public bool ExistingSelected
+        //{
+        //    get
+        //    {
+        //        return mExistingSelected;
+        //    }
+        //    set
+        //    {
+        //        mExistingSelected = value;
+        //        mNewSelected = !value;
+        //    }
+        //}
+
+        private bool mIsSelected;
+        public bool IsSelected
+        {
+            get
+            {
+                return mIsSelected;
+            }
+            set
+            {
+                mIsSelected = value;
+            }
+        }
+
+        private string mName;
+        public string Name
+        {
+            get
+            {
+                return mName;
+            }
+            set
+            {
+                mName = value;
+            }
+        }
+
+        private string mDescription;
+        public string Description
+        {
+            get
+            {
+                return mDescription;
+            }
+            set
+            {
+                mDescription = value;
+            }
+        }
+
+        private string mMatchingAPIName = "";
+        public string MatchingAPIName
+        {
+            get
+            {
+                return mMatchingAPIName;
+            }
+            set
+            {
+                mMatchingAPIName = value;
+            }
+        }
+
+        private ApplicationAPIModel mlearnedAPIl;
+        public ApplicationAPIModel learnedAPI
+        {
+            get
+            {
+                return mlearnedAPIl;
+            }
+            set
+            {
+                mlearnedAPIl = value;
+                mName = value.Name;
+                mDescription = value.Description;
+            }
+        }
+
+        private ApplicationAPIModel mMatchingAPIModel;
+        public ApplicationAPIModel matchingAPIModel
+        {
+            get
+            {
+                return mMatchingAPIModel;
+            }
+            set
+            {
+                mMatchingAPIModel = value;
+                string[] relPathArr = value.ContainingFolderFullPath.Split(new string[] { "API Models" }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
what is that split?? why??

Reply:
coz we're displaying the Matching API's name with Relative Path

Reply:
we have "RelativeFilePath" on repositoryItemBase, can't it be use? i don't like the use of "API Models" because using static strings is never good, see what we do in other places we show relative and if this is best solution

Reply:
It's null.
Now, Updated.